pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -269,36 +270,37 @@ func (i *InfraProvider) Provision(ctx context.Context, dir string, parents asset
 	masterIgnSecret := IgnitionSecret(masterIgnAsset.Files()[0].Data, clusterID.InfraID, "master")
 	machineManifests = append(machineManifests, bootstrapIgnSecret, masterIgnSecret)
 
-	timer.StartTimer(machineStage)
 	// Create the machine manifests.
+	timer.StartTimer(machineStage)
+	machineNames := []string{}
+
 	for _, m := range machineManifests {
 		m.SetNamespace(capiutils.Namespace)
 		if err := cl.Create(ctx, m); err != nil {
 			return fileList, fmt.Errorf("failed to create control-plane manifest: %w", err)
 		}
 		i.appliedManifests = append(i.appliedManifests, m)
+
+		if machine, ok := m.(*clusterv1.Machine); ok {
+			machineNames = append(machineNames, machine.Name)
+		}
 		logrus.Infof("Created manifest %+T, namespace=%s name=%s", m, m.GetNamespace(), m.GetName())
 	}
 
 	{
-		masterCount := int64(1)
-		if reps := installConfig.Config.ControlPlane.Replicas; reps != nil {
-			masterCount = *reps
-		}
-
 		untilTime := time.Now().Add(timeout)
 		timezone, _ := untilTime.Zone()
-		logrus.Infof("Waiting up to %v (until %v %s) for machines to provision...", timeout, untilTime.Format(time.Kitchen), timezone)
+		logrus.Infof("Waiting up to %v (until %v %s) for machines %v to provision...", timeout, untilTime.Format(time.Kitchen), timezone, machineNames)
 		if err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
```

Comment:
Could we run N parallel retries, one for each machine and then wait for all of them to return?

Reply:
Maybe. I played around with creating a wait group to do this, but ended up resolving the issue in a less elegant way. Hopefully this solution is fine for now, and if we want to refactor the checks later we can do that.